### DISCUSSION TITLE
Determine parent of a Sequence

### DISCUSSION THREAD

**Discussion by phillipgsmith** [fo-dicom/fo-dicom#1361] [Q&A] [answered]
Does anyone know of a way to  determine the parent of a Sequence using the interface IDicomDatasetWalker? For example, if I have the sequence "Treatment Session Ion Beam Sequence", its parent is the root. But the parent of "Ion Control Point Delivery Sequence" is "Treatment Session Ion Beam Sequence". Thanks.

**Comment by gofal** [comment, marked as answer]
you have to hold an Stack of DicomItems as private variable in your IDicomDatasetWalker implementation.
Then on OnBeginSequence you know, that you will step a level down, there you push the current item on the stack. And on OnEndSequence you are done with that and can pop the item from your stack.
meanwhile on every item you kan peek the stack for the top item and know the parent.

**Comment by phillipgsmith** [comment]
Thanks so much. I actually figured this out yesterday and tried to delete this discussion but I assume that delete is not an option.

**Reply by mrbean-bremen** [reply]
No need to delete this - it may be helpful to somebody else.

**Reply by phillipgsmith** [reply]
I'm not entirely sure if this would be a good enhancement to the IDicomDatasetWalker but was thinking it would be helpful for EndSequenceItem to have a parameter of type "DicomDataset dataset" that corresponds to OnBeginSequenceItem AND OnEndSequence to have a parameter of type "DicomSequence sequence" that corresponds to OnBeginSequence. This way you would know which sequence and sequence item was ending. Just a thought...
